### PR TITLE
Implemented CLI progres preview

### PR DESCRIPTION
**Pull request by Psyhich**

### CODE REVIEW

File: include/cli_progres.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		void Refresh() noexcept;
+	protected:
+	private:
+		static char ReduceUnit(double &dBytesToReduce) noexcept;
+		static void PrintBytesCell(double dBytesToPrint, char cUnitToPrint) noexcept;
+	private:
+		std::size_t m_nLastAddedLines{0};
+		std::size_t m_nCycleProgress{0};
+
+		const constexpr static std::size_t CYCLE_LIMIT{5000};
```
Comment (VictorApopiy) [inline include/cli_progres.h:28]: define 5000 number

File: include/cli_progres.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		const constexpr static std::size_t SIZE_OF_URI_CELL = 11;
```
Comment (VictorApopiy) [inline include/cli_progres.h:29]: define 11

File: src/main.cpp [hunk 3]
```diff
@@ -1,44 +1,85 @@
...
+			downloaders.GetDownloadProgres();
+
+		CLI::CCLIProgressPrinter progessPrinter;
+		while(!downloaders.IsDone())
+		{
+			progessPrinter.PrintHelp();
+			progessPrinter.PrintProgress(progressData);
+
+			progessPrinter.Refresh();
+			std::this_thread::sleep_for(std::chrono::milliseconds(100));
```
Comment (VictorApopiy) [inline src/main.cpp:75]: define 100

File: src/response.cpp [hunk 4]
```diff
@@ -52,8 +52,8 @@ bool HTTP::CHTTPResponse::LoadHeaders(size_t &nIndex, const std::vector<char> &c
 	bool bIsNewLine{false};
 	bool bIsReadingValue{false};
 
-	std::string sReadKey;
-	std::string sReadValue;
+	std::string sReadKey{""};
+	std::string sReadValue{""};
```
Comment (VictorApopiy) [inline src/response.cpp:56]: String has ctor. You don't need to initialize it with empty data

**Review by VictorApopiy — APPROVED** [review]

File: include/cli_progres.h (hunk 1 above)
Comment (Psyhich) [reply include/cli_progres.h:28]: But it's already defined using `const constexpr static` keywords

File: include/cli_progres.h (hunk 2 above)
Comment (Psyhich) [reply include/cli_progres.h:29]: Same here with keywords

File: src/main.cpp (hunk 3 above)
Comment (Psyhich) [reply src/main.cpp:75]: Defined

File: src/response.cpp (hunk 4 above)
Comment (Psyhich) [reply src/response.cpp:56]: Removed redundant constructor

**Pull request merged**